chore(w3i-rn): Add docs for Web3Inbox React Native SDK (#936)
WalletConnect/walletconnect-docs: docs/web3inbox/usage.mdx

#### Android XML

Adding Web3Inbox to legacy Views should be achieved according to Google's guidelines:
https://developer.android.com/jetpack/compose/migrate/interoperability-apis/compose-in-views.

#### Prevent reloading of Web3Inbox

**IMPORTANT:** If Web3Inbox's inner WebView is reloaded it will lose state and be displayed as blank page.
To prevent reloading of Web3Inbox the activity that contains the `Web3Inbox.View` add
`android:configChanges="orientation|screenSize"` to `<activity>` element `Manifest.xml` that houses Web3Inbox.

</PlatformTabItem>
<PlatformTabItem value="react-native">

This example makes use of [WalletConnect Modal](/2.0/advanced/walletconnectmodal/about?platform=react-native) for simplicity but you can integrate Web3Inbox into your React Native Wallet by providing the following props.

#### Required Props

`projectId`: Get your project ID key on [WalletConnect Cloud](https://cloud.walletconnect.com).

`ethAddress`: Ethereum wallet address.

`onSign`: Web3Inbox client will request user to sign messages with its account private key. The message may be a Sync Storage derivation key or Identity key registration.

> `clientMetadata`: Web3Inbox will consume your metadata for various use use cases, such as displaying your logo and the name of your project.

Reviewer: ```suggestion
`clientMetadata`: Web3Inbox will consume your metadata for various use cases, such as displaying your logo and the name of your project.
```

> `visible` & `setVisible`: Those props are used to handle visibility of Web3Inbox webview.

Reviewer: ```suggestion
`visible` & `setVisible`: These props are used to handle visibility of the Web3Inbox webview.
```

#### Optional Props

`closeButton`: Provide a custom component to close Web3Inbox webview.

`chatEnabled`: Boolean to enable or disable Chat SDK in Web3Inbox (default: `true`), setting it to `false` will also remove the navigation item.

> `pushEnabled`: Boolean to enable or disable Push SDK in Web3Inbox (default: `true`), setting it to `false` will also remove the navigation item.

Reviewer: @devceline I think we will need to relabel and rewire this in the RN SDK too once Notify is merged right?

> `settingsEnabled`: Boolean to enable or disable settigns page in Web3Inbox (default: `true`), setting it to `false` will also remove the navigation item.

Reviewer 2: ```suggestion
`settingsEnabled`: Boolean to enable or disable settings page in Web3Inbox (default: `true`), setting it to `false` will also remove the navigation item.
```
